Doberman/AlarmMonitor.py

```diff
@@ -179,20 +177,18 @@ def check_for_alarms(self):
         db_col = ('logging', 'alarm_history')
         if self.db.count(*db_col, doc_filter) == 0:
             return
-        for doc in self.db.read_from_db(*db_col, doc_filter, sort=[('howbad', -1)]):
-            howbad = int(doc['howbad'])
-            if (howbad,) not in messages:
-                messages[(howbad,)] = []
-            self.db.update_db(*db_col, {'_id': doc['_id']}, updates)
-            messages[(howbad,)].append(doc)
-        if messages:
-            self.logger.warning(f'Found alarms!')
-            for (lvl,), msg_docs in messages.items():
-                message = ''
-                for msg_doc in msg_docs:
-                    time_str = msg_doc['_id'].generation_time.astimezone(None).strftime("%Y-%m-%d %H:%M %Z")
-                    message += f'{time_str}: {msg_doc["msg"]} \n'
-                self.send_message(lvl, message)
+        for doc in self.db._check(*db_col).aggregate([
+            {'$match': doc_filter},
+            {'$group': {
+                '_id': '$howbad',
+                'logged': {'$push': {'$toDate': '$_id'}},
+                'msgs': {'$push': '$msg'}}}]):
+            message = '\n'.join([f'{d.iso_format()}: {m}' for d,m in zip(doc['logged'], doc['msgs'])])
```

Comment:
There is an argument for formatting the message later on (passing the list of timestamps and individual alarms to the separate per-protocol alarm-sending functions) so that it can be done on a per-protocol basis. For example, in a voice call might want to format the time differently, and add pauses between alarms.

Reply:
That's a valid argument but I don't think this feature would be necessary to get this PR merged so we can move onto full-scale testing. We can make a note and come back to this once we've broken the system in and have a better idea of how exactly we would want this to work.